Sources/HttpPipeline/SharedMiddlewareTransformers.swift

```diff
@@ -2,6 +2,34 @@ import Foundation
 import Optics
 import Prelude
 
+public func filterMap<A, B>(
+  _ f: @escaping (A) -> IO<B?>,
```

Comment:
Some food for thought, but I wonder if we should have a `filterMap`-like operation that goes `(Conn<A>) -> IO<B?>`... That way we could recreate some of the transformers below with it.